openedx/core/djangoapps/credit/models.py

```diff
@@ -467,6 +467,28 @@ def add_or_update_requirement_status(cls, username, requirement, status="satisfi
             requirement_status.reason = reason if reason else {}
             requirement_status.save()
 
+    @classmethod
+    @transaction.commit_on_success
+    def remove_requirement_status(cls, username, requirement, status="satisfied"):
+        """
+        Remove credit requirement status for given username.
+
+        Args:
+            username(str): Username of the user
+            requirement(CreditRequirement): 'CreditRequirement' object
+            status(str): Status of the requirement
+
+        """
+
+        requirement_status = cls.objects.filter(
```

Comment:
Can you explain why we are matching on status=status here? That means all calling clients will need to know what the current status is. We might want to make status an optional argument defaulting to None. If status=None when we don't add it to the filtering criteria.

Also, there should really only be one row per username/requirement pair, so this really should be a .get() rather than .filter(). Be sure to add an ItemNotFoundError exception handler (catch, log, and return).